### ISSUE TITLE
Comparing `kSpecialTag` values in `Value::isEqual` might be broken 

### ISSUE THREAD

**Issue by blaugold** [couchbaselabs/fleece#102]
In `Value::isEqual` values which are tagged with `kSpecialTag` are compared through `_byte[1]`:

https://github.com/couchbaselabs/fleece/blob/59ff995b9b2ca8834adbd372f9b213f33af13328/Fleece/Core/Value.cc#L292

But if I understand the memory layout of `Value` correctly, these values only have relevant bits in the first byte.

I think this might have been shadowed by `_byte[1]` usually being zeroed for these values.

**Comment by borrrden** [comment]
The first byte would indicate the type of the value, and the bytes beyond that would be the actual value.  So `_byte[0]` here would be the actual `kSpecialTag` and byte 1 would be its value.

**Comment by blaugold** [comment]
According to the binary layout in `Internal.hh` tags are stored in the upper 4 bits of the first byte and values tagged with `kSpecialTag` use the 2 bits below (also in the first byte) to discriminate the special values:
https://github.com/couchbaselabs/fleece/blob/59ff995b9b2ca8834adbd372f9b213f33af13328/Fleece/Core/Internal.hh#L33

**Comment by snej** [comment]
@blaugold Yeah, that test looks wrong, unless there's something I've forgotten. Values with `kSpecialTag` can be compared by the first byte alone, which is already done at the very start of the `isEqual` method.

I think the test is harmless, though: all Values are 2-byte aligned†, and the padding bytes are zeroed out. So the test will always compare two zero bytes and succeed. But it can be removed.
